crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -333,27 +632,30 @@ impl<'db> CallableBinding<'db> {
 }
 
 /// Binding information for one of the overloads of a callable.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub(crate) struct Binding<'db> {
     /// Return type of the call.
     return_ty: Type<'db>,
 
-    /// Bound types for parameters, in parameter source order.
-    parameter_tys: Box<[Type<'db>]>,
+    /// The formal parameter that each argument is matched with, in argument source order, or
+    /// `usize::MAX` if the argument was not matched to any parameter.
+    argument_parameters: Box<[usize]>,
```

Comment:
I'd prefer an `Option<usize>` over the maigc use of `usize::MAX` or to introduce a newtype wrapper that behaves as `Option<usize>` but does the `magic` encoding internally (e.g. by using an `Option<NonZeroUsize>` internally and that maps the indices by 1 offset).

Reply:
Done